src/griffin/test/contract.cljc

```diff
@@ -107,37 +107,40 @@
   "An implementation of MockState that does not persist"
   []
   (let [s (atom nil)]
-    (reify mock/State
-      (mock/init-state [this i]
+    (reify mock-protocol/State
+      (mock-protocol/init-state [this i]
         (reset! s i)
         this)
-      (mock/swap-state [this f]
+      (mock-protocol/swap-state [this f]
         (swap! s f)
         this))))
 
+(defn var->sym [v]
+  (symbol (str (-> v meta :ns ns-name)) (str (-> v meta :name))))
```

Comment:
[Praise] extracting this in a function makes sense.
[Question] why is the body written differently?

Reply:
The reason I am asking is that I haven't found an authoritative source that would guarantee the metadata or the static fields `ns` and `name` to be present.

The options I can see:
- `(.ns v)` and `(.sym v)` have the advantage that if these were taken away in a subsequent Clojure release this code would break
- `(-> v meta :ns ns-name)` avoids relying on java interop
- [clojure.lang.Var/.toSymbol](https://github.com/clojure/clojure/blob/fb22fd778a272b034684a4ee94509552b46ee8a9/src/jvm/clojure/lang/Var.java#L133) also does the same

Reply:
I'm guessing it's because `(.ns v)` isn't portable to ClojureScript, but `(-> v meta :ns)` is.